app/models/policy/auction.rb

```diff
@@ -0,0 +1,263 @@
+                    'Future'
+                  else
+                    'Open'
+                  end
+    "::Presenter::AuctionStatus::#{status_name}".constantize
+  end
+
+  def status_presenter
+    @status_presenter ||= status_presenter_class.new(self)
+  end
+
+  def user_bid_amount_as_currency(user)
+    number_to_currency(lowest_user_bid_amount)
+  end
+  
+
+  # def current_bid_record
+  #   @current_bid_record ||= bids.sort_by {|bid| [bid.amount, bid.created_at, bid.id] }.first
+  # end
+
+  delegate :label_class, :label, :tag_data_value_status, :tag_data_label_2, :tag_data_value_2,
+           to: :status_presenter
+
+  delegate :start_datetime, :end_datetime, :bids, :bids?, :bid_count, :start_price, :id,
+           :title, :html_summary, :human_start_time, :html_description, :issue_url, :starts_at, :ends_at,
+           :type,
+           to: :auction
+
+  delegate :bidder_name, :bidder_duns_number, :amount, :time, to: :highlighted_bid, prefix: true
+
+  def to_param
+    auction.to_param
+  end
+
+  def lowest_user_bid
+    user_bids.sort_by(&:amount).first || Presenter::Bid::Null.new
+  end
+
+  def lowest_user_bid_amount
+    lowest_user_bid.raw_amount
+  end
+
+  def ==(obj)
+    obj.user == user && obj.auction == auction
+  end
+  
+  protected
+
+  def initialize(auction, user)
+    @auction = presenter_auction(auction)
+    @user = presenter_user(user)
+  end
+
+  def presenter_auction(auction)
+    case auction
+    when Presenter::Auction
+      auction
+    when ::Auction
+      Presenter::Auction.new(auction)
+    when nil
+      nil
+    else
+      fail "Unknown type '#{auction.klass}' for auction"
+    end
+  end
+
+  def presenter_user(user)
+    case user
+    when Presenter::User
+      user
+    when ::User
+      Presenter::User.new(user)
+    when nil
+      nil
+    else
+      fail "Unknown type '#{userklass}' for user"
+    end
+  end
+  
+  def lowest_bids
+    @lowest_bids ||= bids.select {|b| lowest_bid_amount == b.amount }.sort_by(&:created_at).map {|x| Policy::Bid.new(x, self) }
+  end
+
+  def lowest_bid
+    return Presenter::Bid::Null.new if lowest_bids.empty?
+    lowest_bids.first
+  end
+
+  def lowest_bid_amount
+    return nil unless bids?
+    bids.sort_by(&:amount).first.amount
+  end
+
+  def validate_auction
+    unless available?
+      fail UnauthorizedError, 'Auction not available'
+    end
+  end
+  
+  def validate_user
+    unless user_logged_in?
+      fail UnauthorizedError, 'You must be logged in to bid on an auction'
+    end
+
+    unless user_can_bid?
+      fail UnauthorizedError, 'You are not allowed to place a bid in this auction.'
+    end
+  end
+
+  def validate_amount(amount)
+    if amount.to_i != amount
+      fail UnauthorizedError, 'Bids must be in increments of one dollar'
+    end
+
+    if amount > BID_LIMIT
+      fail UnauthorizedError, 'Bid too high'
+    end
+
+    if amount <= 0
+      fail UnauthorizedError, 'Bid amount out of range'
+    end
+  end
```

Comment:
This has become a really long class. Sandi rules cap the class at 100 lines and I have found 75 to be easier on comprehension. Now would be the moment to look for areas of cohesion and start pulling out smaller objects that will likely be reused outside this context.